examples/dreambooth/train_dreambooth_lora_sdxl.py

```diff
@@ -1141,16 +1402,38 @@ def compute_text_embeddings(prompt, text_encoders, tokenizers):
                     model_pred, model_pred_prior = torch.chunk(model_pred, 2, dim=0)
                     target, target_prior = torch.chunk(target, 2, dim=0)
 
-                    # Compute instance loss
-                    loss = F.mse_loss(model_pred.float(), target.float(), reduction="mean")
-
                     # Compute prior loss
                     prior_loss = F.mse_loss(model_pred_prior.float(), target_prior.float(), reduction="mean")
 
+                if args.snr_gamma is None:
+                    loss = F.mse_loss(model_pred.float(), target.float(), reduction="mean")
+                else:
+                    # Compute loss-weights as per Section 3.4 of https://arxiv.org/abs/2303.09556.
+                    # Since we predict the noise instead of x_0, the original formulation is slightly changed.
+                    # This is discussed in Section 4.2 of the same paper.
+                    snr = compute_snr(timesteps)
+                    base_weight = (
+                            torch.stack([snr, args.snr_gamma * torch.ones_like(timesteps)], dim=1).min(dim=1)[
```

Comment:
The code style looks weird here

Reply:
I used the exact same logic that was used here to maintain consistency across examples: https://github.com/huggingface/diffusers/blob/cdadb023a298674239561d0395db2056b4f1d7d7/examples/dreambooth/train_dreambooth.py#L1282

Or are you referring to something else?